scrapy/pipelines/media.py

```diff
@@ -90,6 +103,23 @@ def _process_request(self, request, info):
         )
         return dfd.addBoth(lambda _: wad)  # it must return wad at last
 
+    def _modify_media_request(self, request):
+        httpstatus_list = []
+        if self.handle_httpstatus_list:
+            httpstatus_list = self.handle_httpstatus_list
+        if self.allow_redirects:
+            if not httpstatus_list:
+                httpstatus_list = [i for i in range(1000)
```

Comment:
Maybe we can use a special object to test if an integer is not in some range, something along the lines of 
```
>>> class RangeExclusion(object):
...     def __init__(self, a, b):
...         self.lower_bound = a
...         self.upper_bound = b
...     def __contains__(self, item):
...         return not (self.lower_bound <= item <= self.upper_bound)
... 
>>> 200 in RangeExclusion(300, 400)
True
>>> 301 in RangeExclusion(300, 400)
False
>>> 404 in RangeExclusion(300, 400)
True
```
or using explicit status codes (closer to your suggestion)
```
>>> class ListExclusion(object):
...     def __init__(self, reject_list):
...         self.rejected =  reject_list
...     def __contains__(self, item):
...         return item not in self.rejected
... 
>>> 200 in ListExclusion((301, 302, 303, 307))
True
>>> 301 in ListExclusion((301, 302, 303, 307))
False
>>> 307 in ListExclusion((301, 302, 303, 307))
False
>>> 401 in ListExclusion((301, 302, 303, 307))
True
>>> 
```